pkg/apis/pipeline/v1/taskref_types.go

```diff
@@ -40,3 +40,8 @@ const (
 	// NamespacedTaskKind indicates that the task type has a namespaced scope.
 	NamespacedTaskKind TaskKind = "Task"
 )
+
+// IsCustomTask checks whether the reference is to a Custom Task
+func (tr *TaskRef) IsCustomTask() bool {
+	return tr != nil && tr.APIVersion != "" && tr.Kind != ""
```

Comment:
can `tr.Kind` be anything other than "Task"? I thought this was meant to differentiate between Tasks and ClusterTasks (at least according to the comment in this file)

Reply:
I was also puzzled by this initially, but that is the current behavior (not changing anything in this PR)

Then I realized that I could have a Custom Task with Kind "Task" -- like we do for the Pipelines in Pipelines Custom Task where the Kind is "Pipeline"

Reply:
"ClusterTasks" --> assumed you meant "CustomTask" in this comment

Reply:
so according to the comment in this file and in v1beta1, a Task `kind` can be "Task" or "ClusterTask". I think later on we might have started allowing a Task kind to be any string for use with custom tasks, but the comment doesn't reflect that-- can you update it?

I understand this isn't trying to change existing behavior, but is it possible to have a custom task with apiversion "tekton.dev/v1" and kind "Task"? Could we differentiate between a Task and a CustomTask in this case?

Reply:
Updated the comments

Yes, setting apiversion "tekton.dev/v1" and kind "Task" will create a `CustomRun` instead of a `TaskRun`

We can revisit that behavior in a subsequent PR, will open an issue for further discussion

Note that setting apiversion "tekton.dev/v1" and kind "Pipeline" will create a `CustomRun` which will be handled by the Pipeline-in-Pipeline Custom Task so there's value in using those values for custom functionality

Reply:
Opened https://github.com/tektoncd/pipeline/issues/6457